contract_manager/src/utils/utils.ts

```diff
@@ -0,0 +1,58 @@
+import evmChainsData from "../../store/chains/EvmChains.json";
+import evmPriceFeedContractsData from "../../store/contracts/EvmPriceFeedContracts.json";
+import evmWormholeContractsData from "../../store/contracts/EvmWormholeContracts.json";
+import * as chains from "viem/chains";
+
+export const getEvmPriceFeedContractAddress = (
+  chainId: number,
+): `0x${string}` | undefined => {
+  const chain = evmChainsData.find((c) => c.networkId === chainId);
+  if (!chain) {
+    return undefined;
+  }
+  const contract = evmPriceFeedContractsData.find((c) => c.chain === chain.id);
+  if (!contract?.address || !contract.address.startsWith("0x")) {
+    return undefined;
+  }
+  return contract.address as `0x${string}`;
+};
+
+export const getEvmWormholeContractAddress = (
+  chainId: number,
+): `0x${string}` | undefined => {
+  const chain = evmChainsData.find((c) => c.networkId === chainId);
+  if (!chain) {
+    return undefined;
+  }
+  const contract = evmWormholeContractsData.find((c) => c.chain === chain.id);
+  if (!contract?.address || !contract.address.startsWith("0x")) {
+    return undefined;
+  }
+  return contract.address as `0x${string}`;
+};
+
+export const getAllEvmChainsIds: number[] = evmChainsData.map(
+  (c) => c.networkId,
+);
+
+export const getEvmChainRpcUrl = (chainId: number): string | undefined => {
+  const chain = evmChainsData.find((c) => c.networkId === chainId);
+  if (!chain) {
+    return undefined;
+  }
+
+  // Let's try to use the viem chains without checking if they are working
+  const viemChain = Object.values(chains).find(
+    (c) => c.id === Number(chain.id),
+  );
+  if (viemChain && viemChain.rpcUrls && viemChain.rpcUrls.default) {
+    return viemChain.rpcUrls.default.http[0];
+  }
+
+  // Now let's try to use the json rpc url
+  if (chain.rpcUrl) {
+    return chain.rpcUrl;
+  }
+
+  return undefined;
+};
```

Comment:
Minor nits but for similar reasons as above, I'd write this as:

```suggestion
export const getEvmChainRpcUrl = (chainId: number) => {
  const chain = evmChainsData.find(({ networkId }) => networkId === chainId);
  if (chain === undefined) {
    return undefined;
  } else {
    const viemChain = Object.values(chains).find(({ id }) =>
      id === Number.parseInt(chain.id, 10)
    );
    return viemChain?.rpcUrls?.default?.http?.[0] ?? chain.rpcUrl;
  }
};
```

Things to note here:

1. Using `Number.parseInt` instead of `Number()`.  Generally, I recommend `parseInt` over `Number` because it's more explicit (e.g. what does `Number(something)` actually mean?  With `parseInt(something)`, it's very obvious that I'm parsing a string into an int)
2. Use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) rather than explicit checks, which makes the code far more terse
3. As above, removes early returns